code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/InputBoxPushTemplate.kt

```
@@ -0,0 +1,61 @@
+*/
+
+package com.adobe.marketing.mobile.services.ui.notification.templates
+
+import android.content.Intent
+import com.adobe.marketing.mobile.services.ui.notification.PushTemplateConstants
+import com.adobe.marketing.mobile.util.DataReader
+
+internal class InputBoxPushTemplate : AEPPushTemplate {
+    // Required, the intent action name to be used when the user submits the feedback.
+    internal var inputBoxReceiverName: String?
+
+    // Optional, If present, use it as the placeholder text for the text input field. Otherwise, use the default placeholder text of "Reply".
+    internal var inputTextHint: String?
+        private set
+
+    // Optional, once feedback has been submitted, use this text as the notification's body
+    internal var feedbackText: String?
+        private set
+
+    // Optional, once feedback has been submitted, use this as the notification's image
+    internal var feedbackImage: String?
+        private set
+
+    constructor(data: Map<String, String>) : super(data) {
+        inputBoxReceiverName = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.INPUT_BOX_RECEIVER_NAME, null
+        )
+            ?: throw IllegalArgumentException("Required field \"${PushTemplateConstants.PushPayloadKeys.INPUT_BOX_RECEIVER_NAME}\" not found.")
+        inputTextHint = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.INPUT_BOX_HINT, null
+        )
+        feedbackText = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.INPUT_BOX_FEEDBACK_TEXT, null
+        )
+        feedbackImage = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.INPUT_BOX_FEEDBACK_IMAGE, null
+        )
+    }
+
+    constructor(intent: Intent) : super(intent) {
+        val intentExtras =
+            intent.extras ?: throw IllegalArgumentException("Intent extras are null")
+        inputBoxReceiverName =
```

Since this is required we should throw an exception when this filed does not exist.